app/assets/javascripts/main_menu_flickering.js

```diff
@@ -0,0 +1,39 @@
+//-- copyright
+// OpenProject is a project management system.
+// Copyright (C) 2012-2015 the OpenProject Foundation (OPF)
+//
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the GNU General Public License version 3.
+//
+// OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+// Copyright (C) 2006-2013 Jean-Philippe Lang
+// Copyright (C) 2010-2013 the ChiliProject Team
+//
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the GNU General Public License
+// as published by the Free Software Foundation; either version 2
+// of the License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program; if not, write to the Free Software
+// Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+//
+// See doc/COPYRIGHT.rdoc for more details.
+//++
+
+/* Properly decides main menu expanded state before its drawn
+* Fixes flickering side menu where menu is first expanded, then being
+* collapsed in angular.
+*/
+(function($) {
+  var wrapper = $('#wrapper');
+  if (window.sessionStorage.getItem('openproject:navigation-toggle') === 'collapsed') {
+    $('.can-hide-navigation').addClass('hidden-navigation');
+  }
+  wrapper.show();
+}(jQuery));
```

Comment:
Usually, yes.

In this case this won't work, because then it will be part of the includes in head, and I cannot ensure it is run as soon as the main-menu DOM is loaded.

I thus have to include it manually right after the main-menu DOM items. This originally is also was the reason I didn't use an asset, but rendered the JS directly to avoid a request